src/XMakeTasks/Copy.cs

```diff
@@ -408,6 +409,20 @@ CopyFileWithState copyFile
                 _copiedFiles = new TaskItem[0];
                 return true;
             }
+            
+            if (_destinationFiles != null)
+            {
+                var sameDestinationFiles = _destinationFiles.GroupBy(x => x).Where(g => g.Count() > 1).Select(y => y.First());
```

Comment:
There is one point that is not clear to me. Why are `_destinationFiles[]` are present at this time?? Where is the point where `Copy` constructor is called and `_destinationFiles` is set.